File: ui/src/de/fu_berlin/inf/dpp/ui/model/ValidationResult.java
package de.fu_berlin.inf.dpp.ui.model;

/**
 * This class is used to report the result of a parameter validation back to Javascript. It is
 * serialized to a JSON string.
 */
public class ValidationResult {

  private final boolean valid;

  private final String message;

  /**
   * @param valid true if the validation was successful, false otherwise
   * @param message the message saying why the validation failed
   */
  public ValidationResult(boolean valid, String message) {
    this.valid = valid;
    this.message = message;
  }
}

File: de.fu_berlin.inf.dpp.ui/src/de/fu_berlin/inf/dpp/ui/model/ValidationResult.kt
package de.fu_berlin.inf.dpp.ui.model

/**
 * This class is used to report the result of a parameter validation back to
 * Javascript. It is serialized to a JSON string.
 * @param valid
 * true if the validation was successful, false otherwise
 * @param message
 * the message saying why the validation failed
 */
data class ValidationResult(private val valid: Boolean, private val message: String?)